torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -2501,6 +2510,60 @@ def test_DistributedDataParallel_non_default_stream(self):
                         msg=f"Expected gradient of {expected_grad} but got {avg} on rank {self.rank}",
                     )
 
+        def _test_desync_helper(self, tensor):
+            rank = self.rank
+
+            dist.destroy_process_group()
+
+            # Enable NCCL Async error handling
+            os.environ["NCCL_ASYNC_ERROR_HANDLING"] = "1"
+
+            # Reinitialize global process group with a timeout
+            dist.init_process_group(
+                init_method=INIT_METHOD,
+                backend=BACKEND,
+                world_size=int(os.environ["WORLD_SIZE"]),
+                rank=rank,
+                timeout=timedelta(seconds=2),
+            )
+
+            dist.all_reduce(tensor)
+            self.assertEqual(tensor, torch.tensor([3] * 10, dtype=torch.float32))
+            print('1st all reduce done: Rank ', rank, ' has data ', tensor)
+            if rank != 1:
+                dist.all_reduce(tensor)
+                # Sleep for a small time period so that the ProcessGroup
+                # destructor is not triggered before the asybc error handling
+                # mechanism can throw an exception.
+                time.sleep(5)
+            print('2nd all reduce: Rank ', rank, ' has data ', tensor)
+            # We ensure that the process that is not stuck in the desync exits
+            # cleanly.
+            sys.exit(0)
+
+
+        @unittest.skipIf(
+            BACKEND != "nccl" and BACKEND != "gloo",
+            "Only NCCL and GLOO backend support DistributedDataParallel",
+        )
+        @skip_if_lt_x_gpu(2)
+        @skip_if_rocm
+        def test_DistributedDataParallel_desync(self):
```

Comment:
These tests aren't really DDP tests? Why do we have DistributedDataParallel mentioned here?